### PR TITLE
Include defaultRoles in getRolesByUser() results

### PR DESCRIPTION
**Pull request by developeruz**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #11230

### CODE REVIEW

File: framework/rbac/BaseManager.php [hunk 1]
```diff
@@ -190,6 +190,18 @@ public function getRoles()
     }
 
     /**
+     * Returns defaultRoles as array of Role objects
```
Comment (samdark) [inline framework/rbac/BaseManager.php]: Need `@since 2.0.12`.

Comment (samdark) [inline framework/rbac/BaseManager.php]: Missing `@return`

**Review by samdark — CHANGES_REQUESTED** [review]

File: framework/rbac/BaseManager.php (hunk 1 above)
Comment (developeruz) [inline framework/rbac/BaseManager.php]: Added

**Review by samdark — APPROVED** [review]

**Review by SilverFire — DISMISSED** [review]

File: framework/rbac/BaseManager.php [hunk 2]
```diff
@@ -190,6 +190,20 @@ public function getRoles()
...
+     * @since 2.0.12
+     * @return Role[] default roles. The array is indexed by the role names
+     */
+    public function getDefaultRoles()
+    {
+        $result = [];
+        foreach ($this->defaultRoles as $roleName) {
+            $result[$roleName] = new Role(['name' => $roleName]);
```
Comment (SilverFire) [inline framework/rbac/BaseManager.php]: We already have method `createRole()` for this purpose.
I think it is better to call it instead

```php
$result[$roleName] = $this->createRole($roleName);
```

**Review by SilverFire — CHANGES_REQUESTED** [review]

File: framework/rbac/BaseManager.php (hunk 2 above)
Comment (developeruz) [inline framework/rbac/BaseManager.php]: I changed this line. Thanks.

**Review by SilverFire — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thanks!

**Comment by hirenbhut93** [comment]
@samdark, @developeruz  It's work when define rule with default role? Like http://www.yiiframework.com/doc-2.0/guide-security-authorization.html#using-default-roles

**Comment by samdark** [comment]
If rule is executed before the call — yes.

**Comment by hirenbhut93** [comment]
@samdark What?

**Comment by samdark** [comment]
In that case it's not available at the call time so I think it won't be returned.

File: framework/rbac/BaseManager.php [hunk 3]
```diff
@@ -190,6 +190,20 @@ public function getRoles()
     }
 
     /**
+     * Returns defaultRoles as array of Role objects
+     * @since 2.0.12
+     * @return Role[] default roles. The array is indexed by the role names
+     */
+    public function getDefaultRoles()
```
Comment (cebe) [inline framework/rbac/BaseManager.php]: method name conflicts with the property. This should be changed.

Comment (developeruz) [inline framework/rbac/BaseManager.php]: Should I create a new PR because this one is already merged?
Can you, please, suggest a name for this method?

Comment (arogachev) [inline framework/rbac/BaseManager.php]: @developeruz Yes, you should create new PR. Suggest your variation of naming and we can discuss from there.

**Pull request merged**